src/components/NcInputField/NcInputField.vue

```diff
@@ -440,31 +452,20 @@ export default {
 		pointer-events: none;
 		// Position transition
 		transition: height var(--animation-quick), inset-block-start var(--animation-quick), font-size var(--animation-quick), color var(--animation-quick), background-color var(--animation-quick) var(--animation-slow);
-
-		&--leading-icon {
-			margin-inline-start: var(--default-clickable-area);
-		}
-
-		&--trailing-icon {
-			margin-inline-end: var(--default-clickable-area);
-		}
 	}
 
 	&__input:focus + &__label,
 	&__input:not(:placeholder-shown) + &__label {
-		inset-block-start: -10px;
+		--input-label-font-size: 13px; // minimum allowed font size for accessibility
 		line-height: 1.5; // minimum allowed line height for accessibility
-		font-size: 13px; // minimum allowed font size for accessibility
+		inset-block-start: calc(-1.5 * var(--input-label-font-size) / 2);
```

Comment:
Feels like a magic number, TBH
```suggestion
		inset-block-start: calc(-0.75 * var(--input-label-font-size));
```

Reply:
1.5 is the line height so we get the real line height in pixels (1.5 * font size), to center we need the half ( thats the  / 2) and all of the needs to be negative as we move outside